lib/dry/configurable/dsl.rb

```diff
@@ -26,107 +26,16 @@ def initialize(**options, &block)
-          #
-          # If any additional keyword arguments are provided, e.g.:
-          #
-          #   setting :hash_setting, {my_hash: true}, reader: true
-          #
-          # Then we'll have a `default` of `{my_hash: true}` and an `options` of `{reader:
-          # true}`, which is what we want.
-          #
-          # To work around that first case and ensure our (deprecated) backwards
-          # compatibility holds for Ruby 2.6 and 2.7, we extract all invalid options from
-          # `options`, and if there are no remaining valid options (i.e. if there were no
-          # keyword arguments given), then we can infer the invalid options to be a
-          # default hash value for the setting.
-          #
-          # This approach also preserves the behavior of raising an ArgumentError when a
-          # distinct hash is _not_ intentionally provided as the second positional
-          # argument (i.e. it's not enclosed in braces), and instead invalid keyword
-          # arguments are given alongside valid ones. So this setting:
-          #
-          #   setting :some_setting, invalid_option: true, reader: true
-          #
-          # Would raise an ArgumentError as expected.
-          #
-          # However, the one case we can't catch here is when invalid options are supplied
-          # without hash literal braces, but there are no other keyword arguments
-          # supplied. In this case, a setting like:
-          #
-          #   setting :hash_setting, my_hash: true
-          #
-          # Is parsed identically to the first case described above:
-          #
-          #   setting :hash_setting, {my_hash: true}
-          #
-          # So in both of these cases, the default value will become `{my_hash: true}`. We
-          # consider this unlikely to be a problem in practice, since users are not likely
-          # to be providing invalid options to `setting` and expecting them to be ignored.
-          # Additionally, the deprecation messages will make the new behavior obvious, and
-          # encourage the users to upgrade their setting definitions.
-
-          if Dry::Configurable.warn_on_setting_positional_default
-            Dry::Core::Deprecations.announce(
-              "default value as positional argument to settings",
-              "Provide a `default:` keyword argument instead",
-              tag: "dry-configurable",
-              uplevel: 2
-            )
-          end
-
-          options = {default: invalid_opts}
-        end
-
-        if block && !block.arity.zero?
-          if Dry::Configurable.warn_on_setting_constructor_block
-            Dry::Core::Deprecations.announce(
-              "passing a constructor as a block",
-              "Provide a `constructor:` keyword argument instead",
-              tag: "dry-configurable",
-              uplevel: 2
-            )
-          end
-
-          options = options.merge(constructor: block)
-          block = nil
-        end
-
```

Comment:
Oh boy I am so happy to see this go away

Reply:
Haha, you and me both